Piwik.Tracker/PiwikTracker.cs

```diff
@@ -738,7 +759,7 @@ public TrackingResponse DoBulkTrack()
             }
 
             var postData = new JavaScriptSerializer().Serialize(data);
-            var response = SendRequest(PiwikBaseUrl, "POST", postData, true);
+            var response = await SendRequestAsync(PiwikBaseUrl, "POST", postData, true).ConfigureAwait(false);

```

Comment:
In .NET, await and async doesn't necessary mean asynchronous :gasp: 😲. One reason to use it is of course to make your code asynchronous, by running several Async methods without awaiting, then subsequently awaiting on the Task.

Here, we only have one `SendRequestAsync` method, so there's no asynchronous to talk about. However, async is still useful here. `SendRequest` does a network call, in which traditional synchronous code would have the thread doing nothing, only waiting for the network card to send out the packets, remote server to process, and network card to receive the packets.

`SendRequestAsync` which calls `HttpClient.SendAsync`, would instead release the thread to the threadpool (or continue it's event loop if it's the UI thread) at the `await`. When the packets have been received, the underlying .NET runtime (similar to JVM in Java) would then take a thread from the threadpool and continue the execution, setting our `response` variable.

The advantage is that the freed up threadpool thread can process other things, instead of consuming memory, handles etc but doing nothing

OR

The freed up UI thread can continue processing events, handling button clicks, process WM_PAINT to prevent "freeze", etc